controllers/restore.go

```diff
@@ -768,6 +770,45 @@ func setOptionalProperties(
 			acmRestore.Spec.Hooks.Resources...,
 		)
 	}
+
+	// add any label selector set using the acm restore resource spec
+	if acmRestore.Spec.LabelSelector != nil {
+
+		if veleroRestore.Spec.LabelSelector == nil {
```

Comment:
Just for my understanding: Is there a reason we don't set veleroRestore.Spec.LabelSelector to acmRestore.Spec.LabelSelector?  Is velero or oadp appending their own labelselectors that we then have to modify?

Reply:
acmRestore resource creates the velero restores using the options set on the acmRestore; the velero restore options above were not surfaced to the user so there was no way for a user creating and acm restore resource to set these options for the velero restores

Reply:
Just wondering why we don't do:

```
veleroRestore.Spec.LabelSelector = acmRestore.Spec.LabelSelector 
```

like you do later on with OrLabelSelectors as an example.  This way any existing label selectors will be kept, rather than reconciling to the exact labelSelectors specified in the acmRestore - but perhaps that is desired for some reason?

Reply:
We don't set the veleroRestore.Spec.LabelSelector to acmRestore.Spec.LabelSelector because the operator sets a 
veleroRestore.Spec.LabelSelector when creating the credentials restore file, to exclude resources that should be restored only when the activation step is executed ; we want to keep any predefined veleroRestore.Spec.LabelSelector values and add to those the ones defined by the user using the acmRestore spec

The reason I didn't do the same for the OrLabelSelectors is because we are not using this when creating the restore resource. 
But it makes sense to do the same, to be consistent and for the future, if we decide to use the OrLabelSelectors option, so good point.

I updated the code description to explain why I am appending to the veleroRestore.Spec.LabelSelector and changed the veleroRestore.Spec.OrLabelSelector to append as well.

Reply:
That's another good question : velero restore files once created, they go to completion and cannot be updated.
So the scenario could be : 
1. the user runs a one time acm restore ( syncRestoreWithNewBackups: false ) : this acm restore creates the velero restore files, using the options set by the user. Once this is done, acm restore cannot be modified, the status goes to Finished once all velero restores are completed. Velero restores cannot be updated once created.
2. the user runs a passive acm restore ( syncRestoreWithNewBackups: true, a new set of velero restore will run at the specified interval ). In this case, again Velero restores cannot be updated once created. The user can update the acm restore options ( add / remove labels, etc ) but this gets reflected on the new set of restore files, for the next run. 

I put the PR on hold to fix the issue where the function has too many lines, I'll change this next

Reply:
ah, maybe that makes it simpler in the end if we don't need to update existing velero restores - if we only do this at creation time it becomes simpler (this stuff is so hard to get right) as we just use the user settings at creation time.